Formula/perl-build.rb

```diff
@@ -125,6 +126,15 @@ def install
     %w[perl-build plenv-install plenv-uninstall].each do |cmd|
       (bin/cmd).write_env_script(libexec/"bin/#{cmd}", PERL5LIB: ENV["PERL5LIB"])
     end
+
+    # Some scripts' shebang is set to the realpath of Homebrew perl, which is not
+    # found using `detected_perl_shebang`. Also, since Formula["perl"].bin returns
+    # the opt_bin directory, we need to manually construct the Cellar path.
+    on_linux do
+      inreplace Dir[libexec/"bin/{perl-build,config_data}"] do |s|
+        s.sub! %r{^#!#{HOMEBREW_PREFIX}/Cellar/perl/[^/]+/bin/perl}o, "#!#{Formula["perl"].opt_bin}/perl"
+      end
```

Comment:
**EDIT:** Re-added comment after force-push.

I think the `perl-build` build script replaces the Perl shebang with realpath of Homebrew Perl. I couldn't find another way to deal with this:
- `detected_perl_shebang` will only return system `/usr/` Perl paths
- Trying to inreplace `Formula["perl"].bin` fails as it returns opt_bin:
    ```
    Error: An exception occurred within a child process:
      Utils::Inreplace::Error: inreplace failed
    /home/linuxbrew/.linuxbrew/Cellar/perl-build/1.32/libexec/bin/perl-build:
      expected replacement of #<Pathname:/home/linuxbrew/.linuxbrew/opt/perl/bin> with #<Pathname:/home/linuxbrew/.linuxbrew/opt/perl/bin>
    /home/linuxbrew/.linuxbrew/Cellar/perl-build/1.32/libexec/bin/config_data:
      expected replacement of #<Pathname:/home/linuxbrew/.linuxbrew/opt/perl/bin> with #<Pathname:/home/linuxbrew/.linuxbrew/opt/perl/bin>
    ```